pkg-manager/core/src/install/index.ts

```diff
@@ -262,13 +262,7 @@ export async function mutateModules (
     ? cacheExpired(ctx.modulesFile.prunedAt, opts.modulesCacheMaxAge)
     : true
 
-  if (!maybeOpts.ignorePackageManifest) {
-    for (const { manifest, rootDir } of Object.values(ctx.projects)) {
-      if (!manifest) {
-        throw new Error(`No package.json found in "${rootDir}"`)
-      }
-    }
-  }
+  checkProjectsManifest(ctx, maybeOpts)
```

Comment:
The check should be done in this function: https://github.com/pnpm/pnpm/blob/d9da627cd3b0ec4b4392632ad2ab94866b7a8ad0/workspace/find-workspace-packages/src/index.ts#L11-L25